### PR TITLE
Feat/cypress git action

### PR DESCRIPTION
**Pull request by tobi-bams**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [ ] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by tobi-bams** [comment]
> Oof there are some gnarly tests failing here. Can you take another look tobi?

Yes, sure...

I am working on it!!

**Comment by tobi-bams** [comment]
@ecurrencyhodler all the test are passing now

**Comment by ecurrencyhodler** [comment]
Okay let's get a code review.

File: src/config/ModeDispatcher.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export enum AppMode {
 }
 
 const hosts: { [k: string]: AppMode } = {
-  'localhost:3000': AppMode.TRIBES,
+  'localhost:3000': AppMode.COMMUNITY,
```
Comment (kevkevinpal) [inline src/config/ModeDispatcher.tsx:10]: if you can change this back

File: src/config/host.ts
Comment (kevkevinpal) [inline src/config/host.ts:1]: are we able to make cypress work without needing to edit this file?

File: src/components/common/QR.tsx [hunk 2]
```diff
@@ -59,6 +59,7 @@ export default function QR(props: QRProps) {
           justifyContent: 'center',
           alignItems: 'center'
         }}
+        data-challenge={props.value}
```
Comment (kevkevinpal) [inline src/components/common/QR.tsx:62]: why is this changed?

File: src/components/auth/AuthQR.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { EuiLoadingSpinner } from '@elastic/eui';
 import styled from 'styled-components';
 import { observer } from 'mobx-react-lite';
 import { AuthProps } from 'people/interfaces';
-import { formatRelayPerson } from 'helpers';
+import { formatRelayPerson } from 'helpers/helpers-extended';
```
Comment (kevkevinpal) [inline src/components/auth/AuthQR.tsx:6]: I'm confused why we need to make this change

File: cypress/support/e2e.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        body: JSON.stringify({
+          price_to_meet: 0,
+          description: `Testing out how this works by ${node.alias}`,
+          owner_alias: `${node.alias}`,
+          img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQR9dAUM-b34F_a6DMw8D6fQ_Y0LUIAVzvfCw&usqp=CAU'
+        }),
+        headers: { 'x-user-token': `${node.authToken}` }
+      });
+    } catch (error) {
+      console.log('This is an error');
```
Comment (kevkevinpal) [inline cypress/support/e2e.ts:38]: we might want a better error log here

File: src/config/ModeDispatcher.tsx (hunk 1 above)
Comment (tobi-bams) [reply src/config/ModeDispatcher.tsx:10]: Sure!!

File: src/config/host.ts
Comment (tobi-bams) [reply src/config/host.ts:1]: No, the reason why we have to edit this file is because.
By default according to the code. when the app is running locally the backend is served on port `5002` but stack serves tribes on port `13000`.

That's the reason for the change

File: src/components/common/QR.tsx (hunk 2 above)
Comment (tobi-bams) [reply src/components/common/QR.tsx:62]: We need to log in, and to log in we need to get the `challenge` in Cypress.

This is to enable us to perform the log in function the app does.

File: src/components/auth/AuthQR.tsx (hunk 3 above)
Comment (tobi-bams) [reply src/components/auth/AuthQR.tsx:6]: I will change it back!!

Comment (tobi-bams) [reply src/components/auth/AuthQR.tsx:6]: I just checked why I made this change, It's throwing a typescript error on my machine.

`Module '"helpers"' has no exported member 'formatRelayPerson'.`

File: cypress/support/e2e.ts (hunk 4 above)
Comment (tobi-bams) [reply cypress/support/e2e.ts]: Sure, I just pushed a change for this

File: src/components/auth/AuthQR.tsx (hunk 3 above)
Comment (kevkevinpal) [reply src/components/auth/AuthQR.tsx:6]: thats weird cause helpers exports everything in helpers-extended can you try using the relative path instead of the shortened path?

Comment (tobi-bams) [reply src/components/auth/AuthQR.tsx:6]: I figured the error was because we are importing a `.txs` file inside a `.ts` file, [here](https://github.com/stakwork/sphinx-tribes-frontend/blob/21975b94c30de5e83cafff0416e79f3e2daef751/src/helpers/index.ts#L1)
